src/profile-logic/processed-profile-versioning.js

```diff
@@ -1807,5 +1807,200 @@ const _upgraders = {
       }
     }
   },
+  [38]: (profile) => {
+    // The frame table no longer contains return addresses, it now contains
+    // "nudged" return addresses, i.e. return address minus one byte.
+    // See nudgeReturnAddresses for more details.
+    // The code from nudgeReturnAddresses is duplicated below because this
+    // upgrader needs to stay unaffected by any profile format changes after
+    // version 38.
+    //
+    // This upgrader is needed so that clicking "Re-symbolicate" on old profiles
+    // will obtain correct line numbers and inline frames (once implemented),
+    // and so that the assembly view (once implemented) on an old profile will
+    // assign the correct "total" cost to call instructions.
+    for (const thread of profile.threads) {
+      const samplingSelfStacks = new Set();
```

Comment:
Hmm, I could name them ...V38. Yeah ok, I'll do that. I don't want to extract anything to an external file because external files look like they're heavy and like they're getting in the way, and I don't want to give anyone a reason to remove old upgraders.

Reply:
Actually, then the upgrader code will become longer because those two functions need to access the same tables on the thread. Also, the upgrader code is slightly different by design because it mutates the thread instead of creating new tables.
So I'll leave this as-is after all. I know it's not the most readable code, but it's supposed to be write-once-and-forget code...